src/components/organism/auth/Register/Register.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
+import { getDatabase, ref, set } from 'firebase/database';
+import { validateRegister } from 'helpers/Validate';
+import FacebookIcon from '@material-ui/icons/Facebook';
+import useAuth from 'hooks/useAuth';
+import AuthTemplates from 'templates/AuthTempletes/AuthTemplates';
+
+const InitialFormState = {
+  name: '',
+  surname: '',
+  password: '',
+  email: '',
+};
+
+const Register = () => {
+  const [form, setForm] = useState(InitialFormState);
+  const [error, setError] = useState(false);
+  const [auth, setAuth] = useAuth();
+  const history = useHistory();
+
+  const inputRef = useRef(null);
+
+  const focusInput = () => {
+    inputRef.current.focus();
+  };
+
+  useEffect(() => {
+    focusInput();
+  }, []);
+
+  const resetError = () => {
+    setForm(InitialFormState);
+    setError({});
+  };
+
+  const sendUserData = uid => {
+    const path = `user/${uid}/userDetails`;
+
+    set(ref(getDatabase(), path), {
+      name: form.name,
+      surname: form.surname,
+      email: form.email,
+    })
+      .then(() => {
+        setAuth(true, {
+          email: form.email,
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  };
+
+  const handleSignUp = () => {
+    createUserWithEmailAndPassword(getAuth(), form.email, form.password)
+      .then(userCredential => {
+        sendUserData(userCredential.user.uid);
+      })
+      .catch(err => {
+        switch (err.code) {
+          case 'auth/email-already-in-use':
+          case 'auth/invalid-email':
+            setError({ email: err.message });
+            break;
+          case 'auth/weak-password':
+            setError({ password: err.message });
+            break;
+          default:
+            setError('Something went wrong');
+        }
+      });
+  };
+
+  const handleRegistrationValidation = () => {
+    const errorText = validateRegister(form);
+
+    if (!(Object.keys(errorText).length === 0 && errorText.constructor === Object)) {
+      setError(errorText);
+    } else {
+      resetError();
+      handleSignUp();
+    }
+  };
+  const handleSubmit = e => {
+    e.preventDefault();
+    handleRegistrationValidation();
+  };
+  const updateField = e => {
+    setForm({
+      ...form,
+      [e.target.name]: e.target.value,
+      InitialFormState,
+    });
+  };
+  if (auth) {
+    history.push('/shop');
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
```

Comment:
I like it a lot that we split login and register functions to handle the logic separately! 👏👏👏

We can improve it further by extracting some internal components and reuse them in `Login.js` and `Register.js`. We already have `AuthTemplates` which is great :) Other possible components to extract:

- action buttons row is a perfect candidate to become a separate `FormActions` component
- `const ErrorMessage =  ({ label }) => <p className="error-message">{label}</p>;`
- you're probably already working on it, but just in case: maybe we can use here our `SimpleInput.js` for input fields? :)